var/spack/repos/builtin/packages/ufs-weather-model/package.py

```diff
@@ -94,22 +195,43 @@ def cmake_args(self):
         from_variant = self.define_from_variant
         args = [
             from_variant("32BIT", "32bit"),
-            from_variant("AVX2", "avx2"),
-            from_variant("CCPP", "ccpp"),
+            from_variant("CCPP_32BIT", "ccpp_32bit"),
+            self.define("AVX2", False),  # use target settings from Spack
+            self.define("SIMDMULTIARCH", False),  # use target settings from Spack
+            from_variant("DEBUG", "debug"),
+            from_variant("DEBUG_LINKMPI", "debug_linkmpi"),
             from_variant("INLINE_POST", "inline_post"),
             from_variant("MULTI_GASES", "multi_gases"),
+            from_variant("MOVING_NEST", "moving_nest"),
             from_variant("OPENMP", "openmp"),
             from_variant("PARALLEL_NETCDF", "parallel_netcdf"),
-            from_variant("QUAD_PRECISION", "quad_precision"),
-            from_variant("SIMDMULTIARCH", "simdmultiarch"),
+            from_variant("JEDI_DRIVER", "jedi_driver"),
+            from_variant("CMEPS_AOFLUX", "cmeps_aoflux"),
+            from_variant("APP", "app"),
+            from_variant("CCPP_SUITES", "ccpp_suites").replace(";", ","),
+            from_variant("MPI", "mpi"),
+            from_variant("PDLIB", "pdlib"),
```

Comment:
If the variant name and cmake flag name are always the same, you could do a convenient for-loop like:
```python
args = []
variants = ["debug", "moving_nest", ...]
for variant in variants:
    args.append(from_variant(variant.upper(), variant))
```
But this is completely optional, whatever is easiest for you to understand and maintain.